app/src/main/java/com/artemzin/qualitymatters/network/HostSelectionInterceptor.java

```diff
@@ -0,0 +1,47 @@
+package com.artemzin.qualitymatters.network;
+
+import java.io.IOException;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Such implementation allows us easily change base url in the integration and functional tests!
+ */
+@Singleton
+public class HostSelectionInterceptor implements Interceptor {
+
+    private volatile String baseUrl;
+
+    @Inject
+    public HostSelectionInterceptor() {
+        // Add a no-argument constructor with the @Inject annotation to indicate that Dagger may create instances.
+    }
+
+    @Override
+    public Response intercept(Chain chain) throws IOException {
+        Request request = chain.request();
+        final String host = baseUrl;
+        if (host != null) {
+            final HttpUrl newUrl = request.url()
+                .newBuilder()
+                .host(host)
```

Comment:
It seems I should parse `baseUrl` and build `newUrl` also with `port()` and `scheme()` here